### DISCUSSION TITLE
Thoughts on upgrading hardware from Minisforum GK41 (J4125)?

### DISCUSSION THREAD

**Discussion by Cold-Lemonade** [blakeblackshear/frigate#6902] [Ask A Question]
I have 8 cameras (four 4mp, three 2mp, and one 8mp) and am thinking about adding two or three more. I am currently running Frigate in a docker container on a Minisforum GK41 (J4125) and wonder if I should upgrade to something faster. Any thoughts?

What are your thoughts about intel N5105 based systems? Intel N100? and Intel i3-N305? Beelink makes some attractive units that seem reasonably priced.

**Comment by NickM-27** [comment]
What do the CPU and GPU usage look like? that would be the best indicator

**Reply by Cold-Lemonade** [reply]
According to the system page on my Frigate, my GPU utilization is around 3% and CPU utilization if I add it up across all the cameras and my two usb Coral TPUs is probably 40-60% depending on how much motion is going on.

**Reply by Cold-Lemonade** [reply]
I should add that I'm running the cameras at 5 fps on their lower res substreams for detection and recording 20 fps on the main stream. I realize changing this to 30 fps increases GPU and CPU utilization quite a bit.

**Reply by NickM-27** [reply]
Keep in mind the percentages on the system page for the CPU is % of a single cpu core

**Reply by Cold-Lemonade** [reply]
So I should be able to attach a few more cameras?

**Reply by Cold-Lemonade** [reply]
I ran ```htop``` and its showing that usage of each of the 4 cores is bouncing around 20% to 45%.

**Reply by NickM-27** [reply]
> So I should be able to attach a few more cameras?

I would think so
